packages/components/src/combobox-control/index.tsx

```diff
@@ -151,18 +152,21 @@ function ComboboxControl( props: ComboboxControlProps ) {
 	const matchingSuggestions = useMemo( () => {
 		const startsWithMatch: ComboboxControlOption[] = [];
 		const containsMatch: ComboboxControlOption[] = [];
+		const others: ComboboxControlOption[] = [];
 		const match = normalizeTextString( inputValue );
 		options.forEach( ( option ) => {
 			const index = normalizeTextString( option.label ).indexOf( match );
 			if ( index === 0 ) {
 				startsWithMatch.push( option );
 			} else if ( index > 0 ) {
 				containsMatch.push( option );
+			} else if ( ! shouldFilter ) {
+				others.push( option );
 			}
 		} );
 
-		return startsWithMatch.concat( containsMatch );
-	}, [ inputValue, options ] );
+		return startsWithMatch.concat( containsMatch ).concat( others );
```

Comment:
Currently, when using `shouldFilter={false}` but without providing an alternative list of options, matching items are at the top of the suggested options, while every other item just follows.

![Screenshot 2023-10-30 at 18 22 31](https://github.com/WordPress/gutenberg/assets/1083581/f0f2637a-7016-4557-a98a-688e60823908)

I wonder if, instead, we should disable filtering completely when `shouldFilter={true}` — ie. not looking for a matching suggestion altogether.

I found some relevant examples on the [APG website](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-none/) and on the [ariakit website](https://ariakit.org/components/combobox) showcasing comboboxes with no filtering, where the items order is not affected.

Reply:
Personally, I find it interesting that the items where the label match is first and the ones that contain the string come later, and at the end the ones that match for other criterias. So in that sense I do think the ordering is a good thing here. If you're looking for a parent page, you're more likely to select one where the title matches rather than the description.

Reply:
I think current behavior is fine personally. Maybe it all comes to the name of the prop, which in fact is about including all options or not.

Reply:
I actually like the idea of calling the prop something like `showAllOptions`, it changes the perspective on what this functionality is really about.